File: csp/23/4006. 数组推导/4006.数组推导.py
n = int(input().strip())
b = [0] + list(map(int, input().strip().split()))

cur = 0
max_sum = 0
min_sum = 0
for i in range(1, n + 1):
    if i > 1:
        max_sum += b[i]
        if b[i] > cur:
            cur = b[i]
            min_sum += b[i]

    else:
        cur = b[i]
        max_sum += b[i]
        min_sum += b[i]

print(max_sum)
print(min_sum)

File: csp/23/4006. 数组推导/4006.数组推导.cpp
#include <iostream>
#include <cstring>
#include <cstdio>
#include <algorithm>

using namespace std;

const int N = 110;

int b[N];

int main()
{
    int n, max_sum = 0, min_sum = 0, cur;
    cin >> n;
    for (int i = 1; i <= n; i++)
    {
        cin >> b[i];
        if (i > 1)
        {
            max_sum += b[i];
            if (b[i] > cur)
            {
                min_sum += b[i];
                cur = b[i];
            }
            else
            {
                min_sum += 0;
            }
        }
        else
        {
            cur = b[i];
            max_sum += b[i];
            min_sum += b[i];
        }
    }
    cout << max_sum << endl
         << min_sum << endl;
    return 0;
}